src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -280,8 +280,26 @@ public override void VisitMethod(IMethodSymbol symbol)
                     AddSpace();
                 }
 
+                if (ShouldMethodDisplayReadOnly(symbol))
```

Comment:
>if (ShouldMethodDisplayReadOnly(symbol)) [](start = 16, length = 40)

Please add a test for `ToDisplayString`  that show the conditional behavior for the different parts depending on options.